packages/ai-core/src/browser/os-notification-service.ts

```diff
@@ -0,0 +1,297 @@
+        if (!this.isNotificationSupported()) {
+            return 'denied';
+        }
+
+        if (Notification.permission !== 'default') {
+            return Notification.permission;
+        }
+
+        try {
+            const permission = await Notification.requestPermission();
+            return permission;
+        } catch (error) {
+            console.error('Failed to request notification permission:', error);
+            return 'denied';
+        }
+    }
+
+    /**
+     * Check if OS notifications are supported in the current environment
+     *
+     * @returns true if notifications are supported, false otherwise
+     */
+    isNotificationSupported(): boolean {
+        return typeof window !== 'undefined' && 'Notification' in window;
+    }
+
+    /**
+     * Check if running in Electron environment
+     *
+     * @returns true if running in Electron, false otherwise
+     */
+    isElectronEnvironment(): boolean {
+        return this.isElectron;
+    }
+
+    /**
+     * Show a notification specifically for agent completion
+     * This is a convenience method with pre-configured options for agent notifications
+     *
+     * @param agentName The name of the agent that completed
+     * @param taskDescription Optional description of the completed task
+     * @returns Promise resolving to the notification result
+     */
+    async showAgentCompletionNotification(agentName: string, taskDescription?: string): Promise<OSNotificationResult> {
+        const title = nls.localize('theia/ai-core/agentCompletionTitle', 'Agent Task Completed');
+        const body = taskDescription
+            ? nls.localize('theia/ai-core/agentCompletionWithTask',
+                'Agent "{0}" has completed the task: {1}', agentName, taskDescription)
+            : nls.localize('theia/ai-core/agentCompletionMessage',
+                'Agent "{0}" has completed its task.', agentName);
+
+        return this.showNotification(title, {
+            body,
+            icon: this.getAgentCompletionIcon(),
+            tag: `agent-completion-${agentName}`,
+            requireInteraction: false,
+            data: {
+                type: 'agent-completion',
+                agentName,
+                taskDescription,
+                timestamp: Date.now()
+            }
+        });
+    }
+
+    /**
+     * Detect if running in Electron environment
+     *
+     * @returns true if Electron is detected, false otherwise
+     */
+    private detectElectronEnvironment(): boolean {
+        if (typeof window === 'undefined') {
+            return false;
+        }
+
+        return !!(
+            (window as unknown as { electronAPI?: unknown }).electronAPI ||
+            (window as unknown as { electron?: unknown }).electron ||
+            (window as unknown as { require?: unknown }).require ||
+            (typeof navigator !== 'undefined' && navigator.userAgent && navigator.userAgent.toLowerCase().includes('electron'))
+        );
+    }
```

Comment:
I believe Theia already provides a way to check if you run in Electron.

Reply:
Nice! I adjusted the code accordingly.